libs/agno/agno/agent/agent.py

```diff
@@ -668,18 +690,39 @@ def _run_stream(
         # We track this, so we can add messages after this index to the RunResponse and Memory
         index_of_last_user_message = len(run_messages.messages)
 
-        # 6. Start the Run by yielding a RunStarted event
+        # 1. Start the Run by yielding a RunStarted event
         if stream_intermediate_steps:
             yield self.create_run_response("Run started", session_id=session_id, event=RunEvent.run_started)
 
         # 2. Process model response
-        yield from self._handle_model_response_stream(
+        for event in self._handle_model_response_stream(
             run_response=run_response,
             run_messages=run_messages,
             session_id=session_id,
             response_format=response_format,
             stream_intermediate_steps=stream_intermediate_steps,
-        )
+        ):
+            yield event
+
+        # We should break out of the run function
+        if any(tc.requires_confirmation for tc in run_response.tools or []):
```

Comment:
let's make a helper function for this logic? it looks same in `_run` and in `_run_stream`. For async ones as well.

Reply:
Done